arviz/stats/stats.py

```diff
@@ -719,6 +719,9 @@ def psislw(log_weights, reff=1.0):
     """
     Pareto smoothed importance sampling (PSIS).
 
+    Note that `__sample__` name is reserved only for the dimension storing
```

Comment:
@OriolAbril can you verify if the message is correct... I'm having mixed feelings about it.

Reply:
```suggestion
    Note that `__sample__` name is reserved.
```

Maybe just say this?

Reply:
It is not a reserved name here though. Here inputs _should_ have this dimension as it's the one that will be used. `psislw` can take either a numpy array or an xarray dataarray. 

* if the input if an xarray dataarray and has a `__sample__` dimension, `psislw` will **always** work, independently of the position of the `__sample__` dim, it can be the last, the first, any dimension in the middle.
* if the input is an xarray dataarray withoug a `__sample__` dimension, it is treated as a numpy array, see below.
* if the input is a numpy array, psislw will act on its _last_ dimension, if the input array does not fulfill the convention, psislw won't return the right results. It should still work, but it will act on the wrong dimension.

It's the inputs to the functions that call `psislw` (`loo` and `loo_pit`). 

As a summary, what we should document somewhere, probably in https://github.com/arviz-devs/arviz/issues/1645 or some of the arviz+xarray docs is that ArviZ has 3 reserved dimension names: `chain, draw and __sample__`

Reply:
> if the input array does not fulfill the convention, psislw won't return the right results

I was trying to covey this only, user may assign this name to some other dimension which will lead to broadcast error, the message was far from clear though. Since, `psislw` will be used as a private function only i don't think this message is needed.

> add note in the docstring of psislw about this convention

Should have asked this earlier, what convention were you talking about here?

Reply:
> which will lead to broadcast error,

I expect everything to work actually, but it will return the wrong result, which is even more dangerous.

> what convention were you talking about here?

That `psislw` is applied to the `__sample__` dimension. Take the example and pass both the current log_weights and the transposed version of log weights and you'll see that both work, in xarray one does not care about the order of the dimensions (case 1). Then do the same but when stacking, name the dimension `dim` instead of `__sample__`, now the results differ between original and transposed arrays (case 2). And finally do the same but instead of passing the xarray dataarray to it, use `.values` to pass numpy arrays, here again the result will differ between original and transposed array (case 3)

> Since, psislw will be used as a private function only i don't think this message is needed.

It is not a function the average used would use, but it is a public function, advanced users may use it as they please.

Reply:
Sounds good, I'll update the docstring

Reply:
I've updated the note now